thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/DataAvailabilityTaskScheduler.java

```diff
@@ -95,16 +103,30 @@ public void run() {
           if (isAllDatasetUpdated(detectionConfig, detection2DatasetMap.get(detectionConfig), datasetConfigMap)) {
             if (isWithinSchedulingWindow(detection2DatasetMap.get(detectionConfig), datasetConfigMap)) {
               //TODO: additional check is required if detection is based on aggregated value across multiple data points
-              createDetectionTask(detectionConfig);
+              long endtime = System.currentTimeMillis();
+              createDetectionTask(detectionConfig, endtime);
```

Comment:
Why changing the interface to add endtime?

Reply:
This is just a utility method ensuring both the detection and SLA task have the same end time.

Reply:
It is an overkill to add one additional parameter just for "ensuring both the detection and SLA task have the same end time".
People look at the util function will probably wonder - "why there is only endtime not starttime".